test/distributed/org/apache/cassandra/distributed/impl/INodeProvisionStrategy.java

```diff
@@ -29,137 +29,80 @@
 @Shared(inner = INTERFACES)
 public interface INodeProvisionStrategy
 {
-    enum Strategy
+    @FunctionalInterface
+    interface Factory
```

Comment:
this lets me control this interface a lot more in my test... right now the code assumes node1 never goes away, but my test invalidates that assumption... so seed selection and more must live with my test